lib/tasks/whitehall_importer.rb

```diff
@@ -120,37 +127,93 @@ def create_edition(document, translation, whitehall_edition, edition_number)
         created_at: whitehall_edition["created_at"],
       )
 
-      Edition.create!(
+      edition = Edition.create!(
         document: document,
         number: edition_number,
         revision_synced: true,
         revision: revision,
-        status: Status.new(
-          state: state(whitehall_edition),
-          revision_at_creation: revision,
-        ),
+        status: initial_status(whitehall_edition, revision),
         current: whitehall_edition["id"] == most_recent_edition["id"],
         live: live?(whitehall_edition),
         created_at: whitehall_edition["created_at"],
         updated_at: whitehall_edition["updated_at"],
         created_by_id: user_ids[first_author["whodunnit"]],
         last_edited_by_id: user_ids[last_author["whodunnit"]],
       )
+
+      set_current_status(whitehall_edition, edition) if whitehall_edition["state"] == "withdrawn"
+    end
+
+    def initial_status(whitehall_edition, revision)
+      author = if whitehall_edition["state"] == "withdrawn"
+                 published_author(whitehall_edition)
+               else
+                 current_author(whitehall_edition)
+               end
+
+      raise AbortImportError unless author
+
+      Status.new(
+        state: state(whitehall_edition),
+        revision_at_creation: revision,
+        created_by_id: user_ids[author["whodunnit"]],
+        created_at: author["created_at"],
+      )
+    end
+
+    def set_current_status(whitehall_edition, edition)
+      author = current_author(whitehall_edition)
+
+      raise AbortImportError unless valid_state?(whitehall_edition)
```

Comment:
I think this should also have a message to explain why it's aborted. I'm a bit confused why this doesn't check author where the other method does?

Reply:
If there isn't an author you'd never get as far as calling this method. So but this point there will always be at least one author, it's just that there might only be one and current_author might be the same as the published_author if there's weird data in the export.

Reply:
Added a message and check for an unpublishing.